objects/Door.js

```diff
@@ -0,0 +1,74 @@
+const DOOR_TRIGGERS = {
+    AUTOMATIC: 'auto',
+    UP: 'up',
+    DOWN: 'down'
+}
+class Door extends FinishFlag {
+    constructor(x, y, tileSize, type, tilemapHandler, extraAttributes = {}) {
+        super(x, y, tileSize, type, tilemapHandler, extraAttributes);
+        
+        // overrides
+        this.lockedSpriteIndex = SpritePixelArrays.getIndexOfSprite(ObjectTypes.LOCKED_DOOR);
+        this.lockedSpriteYPos = this.lockedSpriteIndex * this.tileSize;
+        this.sound = SoundHandler.door;
```

Comment:
i wouldn't save the sound in a variable. it's like 100kb, and a bunch of doors means, every door instance needs to hold that in it's memory. i would save SoundsHandler.door.key f.e. in the variable and when you need to play it, you can use
SoundHandler[instance.sound].stopAndPlay();
hope i made that clear
and same goes for FinishFlag.js

Reply:
Doesn’t this just store a reference to the same Sound object that `SoundHandler.door` references? I don't think it would make a new copy each time. 

You can test this out in the console:
```
const w1 = SoundHandler.win;
const w2 = SoundHandler.win;
console.log(w1.loaded); // prints: true
console.log(w2.loaded); // prints: true
w1.loaded = ":)"; // should mutate the object that both w1 and w2 refer to
console.log(w1.loaded); // prints: ":)"
console.log(w2.loaded); // prints: ":)"
```